internal/expr/parser.go

```diff
@@ -206,17 +219,38 @@ loop:
 		switch p.input[p.pos-1] {
 		// If the preceding byte is one of these then we might be at the start
 		// of an expression.
-		case ' ', '\t', '\n', '\r', '=', ',', '(', '[', '>', '<', '+', '-', '*', '/', '|', '%':
+		case ' ', '\t', '\n', '\r', '=', ',', '[', '>', '<', '+', '-', '*', '/', '|', '%':
+			break loop
+		case '(':
+			p.bracketLevel++
+			break loop
+		case ')':
+			p.bracketLevel--
 			break loop
 		}
 	}
 
-	p.skipBlanks()
-
+	p.partStart = p.pos
 	return nil
 
 }
 
+// startSubpart is called to allow parsing of nested parts.
+func (p *Parser) startSubpart() *checkpoint {
```

Comment:
This comment feels slightly off to me, it seems like a comment I'd expect on a call itself rather than the method definition. I think it's worth mentioning the checkpoint, perhaps something like this?
```go
// startSubpart creates a checkpoint and prepares to parse a nested part.
```
Also, given that the sub-part checkpoints are handled slightly differently (when restoring, the new position is computed rather than being a simple replacement), is it worth adding a new `subpartCheckpoint` type here? This would also match the use of `scp` rather than `cp` so it could feel natural

Reply:
Changed the comment. I did have a separate type for it at some point yesterday but with the current implementation you can actually use the checkpoint it returns as a regular checkpoint. I've changed the use of it in `parseFunc` to reflect this as well.